src/zenml/orchestrators/step_launcher.py

```diff
@@ -259,7 +344,9 @@ def _bypass() -> None:
                                 force_write_logs=force_write_logs,
                             )
                             break
-                        except BaseException as e:  # noqa: E722
+                        except RunStoppedException as e:
+                            raise e
+                        except Exception as e:  # noqa: E722
```

Comment:
Why not `BaseException` anymore, which also covers keyboard interrupt?

Reply:
Mhm only the last retry executes the failure hook, I'm not sure how to best handle this tbh. I would just leave it as is as I haven't heard any complaints about this tbh.